### PR TITLE
feat: add support for Webpack 5

### LINKED ISSUE #142: Support webpack v5+
Hi,

Bundling with Webpack v5+ produces the following error: `Module not found: Error: Can't resolve 'path' in <path-to-universal-module>`, because Webpack doesn't polyfill NodeJS module anymore.

PR is incoming.

EDIT (off topic):

@ScriptedAlchemy, what about centralizing `react-universal-component`, `webpack-flush-chunks`, and this package, in a monorepo located eg. at `faceyspacey/react-universal`? It would be an opportunity to start with a clean state, to reduce dependencies and maintenance cost, and I think it will favor contributions.

### PR DESCRIPTION
**Pull request by cdoublev**
Webpack 5 doesn't polyfill NodeJS modules anymore. This change replaces usage of NodeJS `path` with
`path-browserify`.

fix #142

### CODE REVIEW

**Comment by twelve17** [comment]
@cdoublev may I ask why you closed this PR?

**Comment by cdoublev** [comment]
I believe there will be no more support (including merging PR) for this package from the author or the current maintainer, as well as for the other two related packages: `react-universal` and `webpack-flush-chunks`.

I forked and cleaned these 3 packages. Doing this for `react-universal` makes me realize that this plugin currently has a lot of uncovered edge cases for detecting/transforming `import(component)`. It covers the main cases but since you can use `react-universal` without it, it might not be worth using it. A new collaborator might be confused to see an asynchronous `import(component)` executed at runtime if nobody tells him it is transformed for a client/server differentiated execution. And if you remove the extra features from `react-universal` and` webpack-flush-chunks` that you don't need, you may end up with an amount of code that may be not worth to release as a package.

React is not moving so fast on this, but eventually server side components may also makes this packages family useless. 

To sum up, I believe it's not worth investing more time on them.

**Comment by twelve17** [comment]
Thank you for this thorough explanation! 👍

**Comment by webdevnerdstuff** [comment]
> I believe there will be no more support (including merging PR) for this package from the author or the current maintainer, as well as for the other two related packages: `react-universal` and `webpack-flush-chunks`.

@cdoublev Would you please add something to the docs saying that this package will be depreciated as of Webpack 5?

**Comment by cdoublev** [comment]
I don't own the rights to merge a commit. You can try to create a new issue in the main package (`react-universal-component`), tag the author/maintainer, and ask for adding a deprecation note and/or a call period for new maintainers and contributors. I'll help as much as possible as a contributor.

**Comment by ScriptedAlchemy** [comment]
Agreed. I’ve invested in server-side suspense renderrers and no longer can afford the time to maintain the universal OSS suite. Most of my work is going into module federation
